src/test/java/DeadLinksTest.java

```diff
@@ -0,0 +1,45 @@
+import java.io.*;
+import org.junit.*;
+import org.junit.runner.*;
+import org.junit.runners.*;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Java code implementing the below contract;
+ Test calling main(String[] args) and verifying the program system output correctness.
+ */
+
+@RunWith(JUnit4.class)
+public class DeadLinksTest {
+
+  private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+  private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+
+  @Before
+  public void setUpStreams() {
+    System.setOut(new PrintStream(outContent));
+    System.setErr(new PrintStream(errContent));
+  }
+
+  @Test
+  public void out() {
+    String expectedResult = "{\"urlForCheck\":\"http://roofing.tilda.ws\",\"err404\":{\"size\":1,\"urls\":[\"http://roofing.tilda.ws/somepage.html\"]},\"err50x\":{\"size\":0,\"urls\":[]},\"dead\":1,\"total\":14}";
```

Comment:
@VAndreyD please follow the exact requested format. Keys are `url`, `404`, `50x`.